Mathlib/MeasureTheory/Measure/Typeclasses.lean

```diff
@@ -494,7 +494,39 @@ instance isFiniteMeasure_sFiniteSeq [h : SFinite μ] (n : ℕ) : IsFiniteMeasure
 lemma sum_sFiniteSeq (μ : Measure α) [h : SFinite μ] : sum (sFiniteSeq μ) = μ :=
   h.1.choose_spec.2.symm
 
+/-- A countable sum of finite measures is s-finite. -/
+lemma sfinite_sum_of_countable {ι : Type*} [Countable ι]
+    (m : ι → Measure α) [∀ n, IsFiniteMeasure (m n)] : SFinite (Measure.sum m) := by
+  classical
+  obtain ⟨f, hf⟩ : ∃ f : ι → ℕ, Function.Injective f := Countable.exists_injective_nat ι
+  let e : ι ≃ range f := Equiv.ofInjective f hf
+  let m' : ℕ → Measure α := fun n ↦ if hn : n ∈ range f then m (e.symm ⟨n, hn⟩) else 0
```

Comment:
I see 2 ways to move parts of this proof to reusable lemmas.

* Add `Measure.sum_comp_equiv`, instances for `IsFiniteMeasure (∑ i, f i)` and `[Finite ι] : IsFiniteMeasure (.sum f)`, then treat `[Finite ι]` and `[Infinite ι]` separately.
* Add `Measure.sum_extend_zero` using `Function.extend` instead of the equivalent definition you use, then apply it.

Maybe, we should have all these supporting lemmas/instances.

Reply:
I have added `Measure.sum_comp_equiv` and `Measure.sum_extend_zero`. Indeed, the proof becomes nicer with these! On the other hand, I don't think that splitting into finite or infinite \iota would make things nicer.

Reply:
I generalized `sum_extend_zero` and `Function.apply_extend`.
bors d+